plugins/QnA/class.qna.plugin.php

```diff
@@ -553,31 +548,36 @@ public function discussionController_qnAOptions_create($sender, $discussionID =
      *
      * @param array|object $discussion The discussion to recalculate.
      * @param bool $return Whether to return the result or update the discussion.
+     * @return array $set (optional based on the $return argument). Discussion QnA data.
      */
     public function recalculateDiscussionQnA($discussion, bool $return = false) {
-        // Find comments in this discussion with a QnA value.
         $set = [];
 
-        $row = Gdn::sql()->getWhere('Comment',
-            ['DiscussionID' => val('DiscussionID', $discussion), 'QnA is not null' => ''], 'QnA, DateAccepted', 'asc', 1)->firstRow(DATASET_TYPE_ARRAY);
+        // Look for at least one accepted answer/comment.
+        $acceptedComment = Gdn::sql()->getWhere('Comment',
+            ['DiscussionID' => val('DiscussionID', $discussion), 'QnA' => 'Accepted'], '', 'asc', 1)->firstRow(DATASET_TYPE_ARRAY);
 
-        if (!$row) {
-            if (val('CountComments', $discussion, 0) > 0) {
+        if ($acceptedComment) {
+            $set['QnA'] = 'Accepted';
+            $set['DateAccepted'] = $acceptedComment['DateAccepted'];
+            $set['DateOfAnswer'] = $acceptedComment['DateInserted'];
+        } else {
+            // Look for at least one untreated answer/comment.
+            $answeredCommment = Gdn::sql()->getWhere('Comment',
+                ['DiscussionID' => val('DiscussionID', $discussion), 'QnA is null' => ''], '', 'asc', 1)->firstRow(DATASET_TYPE_ARRAY);
+            if ($answeredCommment) {
                 $set['QnA'] = 'Answered';
+                $set['DateAccepted'] = null;
+                $set['DateOfAnswer'] = null;
```

Comment:
If the comment was answered should we be setting the DateOfAnswer here?

Reply:
No we shouldn't. I removed it in the lastest commit.